### PR TITLE
feat: add userAgent property to Navigator's prototype

### PR DESCRIPTION
**Pull request by randomicon00**
PR is related to #14362 

1. Added user_agent field to `BoostrapOptions` struct.
2. Added `userAgent` prototype's property to class Navigator following the same approach as  for `gpu`. 
3. Added `userAgent` variable that gets initialized when `MainWorker::from_bootsrap_options` gets invoked. 
4. Extracted `userAgent`, and renamed to `userAgentInfo` in order to assign it to the navigator's prototype when `mainRuntime` function gets executed.

This is a draft and I didn't include the tests yet. I would like to know if this is the right approach. What makes me doubt is the fact that `WorkerOptions` also contains a `user_agent` field, but I didn't explore if there was a way to get access to it in the JavaScript code. 

I am open to your suggestions. Thank you.

### CODE REVIEW

**Comment by lucacasonato** [comment]
@randomicon00 The approach looks good. I think you can remove the second `user_agent` from `WorkerOptions` (the one outside of `BootstrapOptions`), and update all uses to use `boostrap.user_agent` instead.

**Comment by randomicon00** [comment]
@lucacasonato & @bartlomieju sure. Will take into account and apply your suggestions.

**Comment by bartlomieju** [comment]
@randomicon00 I would love to land this feature for v1.22, which is to be released next week. Do you need any help with the PR?

**Comment by randomicon00** [comment]
There is just regression error that was introduced that is blocking me right now. The code used to work. I should be able to make an update by Wednesday.

**Comment by bartlomieju** [comment]
> There is just a regression error that was introduced that is blocking me right now. The code used to work. I should be able to make an update by Wednesday.

Okay, let me know if you need any help

**Comment by randomicon00** [comment]
I did the first part of moving `user_agent` to `BootstrapOptions`. I will write tests in the next commit. Let me know please if there are any specific tasks I need to do on top of that. 
Thank you.

**Comment by randomicon00** [comment]
@bartlomieju I have to understand tomorrow why `userAgent`is not defined on `Navigator`causing the test to fail.

**Comment by randomicon00** [comment]
Ready to be reviewed. Thanks.

File: cli/main.rs [hunk 1]
```diff
@@ -162,14 +162,15 @@ fn create_web_worker_callback(ps: ProcState) -> Arc<CreateWebWorkerCb> {
...
+        user_agent: version::get_user_agent(),
       },
       extensions,
       unsafely_ignore_certificate_errors: ps
         .flags
         .unsafely_ignore_certificate_errors
         .clone(),
       root_cert_store: ps.root_cert_store.clone(),
-      user_agent: version::get_user_agent(),
+      //user_agent: version::get_user_agent(),
```
Comment (bartlomieju) [inline cli/main.rs:173]: Remove

File: cli/main.rs [hunk 2]
```diff
@@ -263,14 +264,15 @@ pub fn create_main_worker(
...
+      user_agent: version::get_user_agent(),
     },
     extensions,
     unsafely_ignore_certificate_errors: ps
       .flags
       .unsafely_ignore_certificate_errors
       .clone(),
     root_cert_store: ps.root_cert_store.clone(),
-    user_agent: version::get_user_agent(),
+    //user_agent: version::get_user_agent(),
```
Comment (bartlomieju) [inline cli/main.rs:275]: Remove

File: cli/standalone.rs [hunk 3]
```diff
@@ -286,9 +286,10 @@ pub async fn run(
       runtime_version: version::deno(),
       ts_version: version::TYPESCRIPT.to_string(),
       unstable: metadata.unstable,
+      user_agent: version::get_user_agent(),
     },
     extensions: ops::cli_exts(ps.clone(), true),
-    user_agent: version::get_user_agent(),
+    //user_agent: version::get_user_agent(),
```
Comment (bartlomieju) [inline cli/standalone.rs:292]: Remove

File: runtime/examples/hello_runtime.rs [hunk 4]
```diff
@@ -39,11 +39,12 @@ async fn main() -> Result<(), AnyError> {
       runtime_version: "x".to_string(),
       ts_version: "x".to_string(),
       unstable: false,
+      user_agent: "hello_runtime".to_string(),
     },
     extensions: vec![],
     unsafely_ignore_certificate_errors: None,
     root_cert_store: None,
-    user_agent: "hello_runtime".to_string(),
+    //    user_agent: "hello_runtime".to_string(),
```
Comment (bartlomieju) [inline runtime/examples/hello_runtime.rs:47]: Remove

File: runtime/web_worker.rs [hunk 5]
```diff
@@ -312,7 +312,7 @@ pub struct WebWorkerOptions {
   pub extensions: Vec<Extension>,
   pub unsafely_ignore_certificate_errors: Option<Vec<String>>,
   pub root_cert_store: Option<RootCertStore>,
-  pub user_agent: String,
+  //pub user_agent: String,
```
Comment (bartlomieju) [inline runtime/web_worker.rs:315]: Remove

File: runtime/worker.rs [hunk 6]
```diff
@@ -48,7 +48,7 @@ pub struct WorkerOptions {
   pub extensions: Vec<Extension>,
   pub unsafely_ignore_certificate_errors: Option<Vec<String>>,
   pub root_cert_store: Option<RootCertStore>,
-  pub user_agent: String,
+  //pub user_agent: String,
```
Comment (bartlomieju) [inline runtime/worker.rs:51]: Remove

File: runtime/worker.rs [hunk 7]
```diff
@@ -364,9 +364,10 @@ mod tests {
         runtime_version: "x".to_string(),
         ts_version: "x".to_string(),
         unstable: false,
+        user_agent: "x".to_string(),
       },
       extensions: vec![],
-      user_agent: "x".to_string(),
+      //user_agent: "x".to_string(),
```
Comment (bartlomieju) [inline runtime/worker.rs:370]: Remove

**Review by bartlomieju — CHANGES_REQUESTED** [review]

File: runtime/worker.rs (hunk 7 above)
Comment (randomicon00) [reply runtime/worker.rs]: Done!

File: runtime/web_worker.rs (hunk 5 above)
Comment (bartlomieju) [reply runtime/web_worker.rs:315]: @randomicon00 this is still not addressed

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you @randomicon00, nice work!

**Pull request merged**